### PR TITLE
Add API endpoint for pull-from-upstream runs

### PR DESCRIPTION
**Pull request by lbarcziova**
Related to packit/dashboard#199

TODO:

- [x] Write new tests or update the old ones to cover new functionality.
- [x] Update doc-strings where appropriate.

---

RELEASE NOTES BEGIN
￼
RELEASE NOTES END

### CODE REVIEW

File: packit_service/service/api/propose_downstream.py [hunk 1]
```diff
@@ -74,23 +54,16 @@ class ProposeResult(Resource):
...
-        if not dowstream_pr:
+        if (
+            not sync_release_target_model
+            or sync_release_target_model.sync_release.job_type
+            != SyncReleaseJobType.propose_downstream
+        ):
             return response_maker(
                 {"error": "No info about propose downstream target stored in DB"},
                 status=HTTPStatus.NOT_FOUND,
             )
```
Comment (lachmanfrantisek) [inline packit_service/service/api/propose_downstream.py:67]:
```suggestion
            return response_maker(
                {"error": "No info about target release sync stored in DB"},
                status=HTTPStatus.NOT_FOUND,
            )
```

File: packit_service/service/api/pull_from_upstream.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+from packit_service.service.api.parsers import indices, pagination_arguments
+from packit_service.service.api.utils import (
+    response_maker,
+    get_sync_release_target_info,
+    get_sync_release_info,
+)
+
+logger = getLogger("packit_service")
+
+ns = Namespace("pull-from-upstream", description="Pull from upstream")
```
Comment (lachmanfrantisek) [inline packit_service/service/api/pull_from_upstream.py:23]: Do you think we should differentiate between `pull-from-upstream` and  `propose-downstream` even if we use the same model for those?

But I agree that they will probably diverge even more in the future...

(= Alternatively, we can create a new `sync-release` endpoint, but I am really not sure here..;)

**Review by lachmanfrantisek — APPROVED** [review]
:+1:

**Review by jpopelka — APPROVED** [review]
:+1:

File: packit_service/service/api/pull_from_upstream.py (hunk 2 above)
Comment (lbarcziova) [reply packit_service/service/api/pull_from_upstream.py:23]: > But I agree that they will probably diverge even more in the future...

Yes, that's the reason why I think it makes sense to have separate endpoints.

**Pull request merged**